arch/arm/core/aarch32/cortex_m/mpu/arm_mpu.c

```diff
@@ -15,6 +15,16 @@
 #include <logging/log.h>
 LOG_MODULE_DECLARE(mpu);
 
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+#define MPU_NODEID DT_INST(0, arm_armv7m_mpu)
```

Comment:
I think DT_NODELABEL(mpu) would work instead of ifdeffing things here too.

Reply:
preference to utilize DT_INST/compat over DT_NODELABEL.